### PR TITLE
Lesson 1.6

### PR DESCRIPTION
**Pull request by viniciuspietscher**

### CODE REVIEW

File: src/AddTodoForm.js [hunk 1]
```diff
@@ -19,14 +20,12 @@ function AddTodoForm({ onAddTodo }) {
...
-        name='title'
-        value={todoTitle}
-        onChange={handleTitleChange}
-      />
+      <InputWithLabel
+        todoTitle={todoTitle}
+        handleTitleChange={handleTitleChange}
+      >
+        Title:
+      </InputWithLabel>
```
Comment (kwilson05) [inline src/AddTodoForm.js]: Looking good! 👍 

You're only missing the `id` prop here. That will allow two instances of the `InputWithLabel` to be created. Also, you can still pass in an `isFocused` prop to so the component knows if it should autoFocus

File: src/App.js [hunk 2]
```diff
@@ -17,6 +17,12 @@ function useSemiPersistentState() {
 function App() {
   const [todoList, setTodoList] = useSemiPersistentState()
 
+  function handleRemoveTodo(id) {
+    const newTodoList = todoList.filter((todoItem) => id !== todoItem.id)
+
+    setTodoList(newTodoList)
+  }
```
Comment (kwilson05) [inline src/App.js:24]: awesome! I used `todoList.filter` too 🎖️

File: src/InputWithLabel.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+function InputWithLabel({
+  todoTitle,
+  type = "text",
+  handleTitleChange,
+  children,
+}) {
+  const inputRef = useRef()
+  useEffect(() => {
+    inputRef.current.focus()
+  })
```
Comment (kwilson05) [inline src/InputWithLabel.js]: Figure out how to use the `isFocused` prop in this method. The road to react book has a good example of it.

File: src/InputWithLabel.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  return (
+    <>
+      <label htmlFor={todoTitle}>{children}</label>
+      <input
+        ref={inputRef}
+        type={type}
+        id={todoTitle}
+        value={todoTitle}
+        onChange={handleTitleChange}
+      />
```
Comment (kwilson05) [inline src/InputWithLabel.js]: You should use that `id` prop here when you pass it in

File: src/TodoListItem.js [hunk 5]
```diff
@@ -1,5 +1,10 @@
-function TodoListItem({ todo }) {
-  return <li>{todo.title}</li>
+function TodoListItem({ todo, onRemoveTodo }) {
+  return (
+    <li>
+      {todo.title}
+      <button onClick={() => onRemoveTodo(todo.id)}>Remove</button>
```
Comment (kwilson05) [inline src/TodoListItem.js:5]: Nice nice!! 💯

**Review by kwilson05 — CHANGES_REQUESTED** [review]
Looking good, Vini! 👍 

I'd like you to make the changes I suggested then put the PR up for review again before merging. I trust you figure out the changes but want to make sure you're a good state before starting the next exercise 😄 
Holler at me if anything

File: src/AddTodoForm.js (hunk 1 above)
Comment (viniciuspietscher) [reply src/AddTodoForm.js]: I'm not sure which `id` would be that, is the `todoTitle.id` ?

Edit: ohh nvm, just found it on the book! :+1:

File: src/TodoListItem.js (hunk 5 above)
Comment (viniciuspietscher) [reply src/TodoListItem.js:5]: woot woot! :+1:

File: src/AddTodoForm.js (hunk 1 above)
Comment (kwilson05) [reply src/AddTodoForm.js]: Nice! That's what programming is about. Reading documentation ahahah

File: src/InputWithLabel.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  children,
+  isFocused,
+}) {
+  const inputRef = useRef()
+  useEffect(() => {
+    if (isFocused && inputRef.current) {
+      inputRef.current.focus()
+    }
+  })
+
```
Comment (kwilson05) [inline src/InputWithLabel.js:17]: 🎖️

File: src/InputWithLabel.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  return (
+    <>
+      <label htmlFor={todoTitle}>{children}</label>
+      <input
+        ref={inputRef}
+        type={type}
+        id={id}
```
Comment (kwilson05) [inline src/InputWithLabel.js:24]: yup yup!! 💯

**Review by kwilson05 — APPROVED** [review]
stamp of approval! 💥

**Pull request merged**